### DISCUSSION TITLE
VSCode %%SAS Magic in notebook

### DISCUSSION THREAD

**Discussion by aplunket** [sassoftware/saspy#418] [Q&A] [answered]
When I run SAS code in a VSCode jupyter notebook via the %%SAS magic command the log that gets outputted when the code runs is changing whole notebook to the SAS output theme (from the standard VSCode dark mode theme). 

I assume that its due to the ods html5 statement setting the style to "HTMLBlue".

Is there anyway I can stop this from happening?

**Comment by tomweber-sas** [comment, marked as answer]
First, I don't have the answer, sorry. 
Yes, I expect it's some interaction w the UI and the HTML, but I'm neither a UI nor html guy.
I remember that the jupyter notebook version, years ago, did the same kind of thing. It doesn't happen in Lab, the newer jupyter UI.

One question, if instead of using %%SAS, if you use saspy's submitLST() does it still happen? Just trying to get more data points.
sas.submitLST('data a; x=1; run;')  or whatever code you were running, does basically the same thing that the %%SAS magic does, but it's a different code path, so curious to see if there's any difference.

Thanks,
Tom

**Reply by aplunket** [reply]
Thanks for the reply Tom!

Interestingly using submitLST() it doesn't happen. 

As you mentioned testing this on jupyter lab and seems to work fine so will just instruct people to use that instead.

**Reply by tomweber-sas** [reply]
Cool, then there is something different between the two paths. I don't really know what the magic path could be doing, so yes, using the submitLST() method would be the quick answer. The magic things are only jupyter constructs too, so those aren't part of saspy, and if you run your saspy code in any other Python environment, the magic wouldn't even work, whereas submitLST() is part of saspy, so it works in any python environment there is. 
Hopefully that works for you and your folks!

Thanks,
Tom
